dsc/workflows/opencourseware.py

```diff
@@ -0,0 +1,158 @@
+        """Yield source metadata from metadata JSON file in zip archive.
+
+        This method expects a JSON file called "data.json" at the root
+        level of the the zip file.
+        """
+        with smart_open.open(file, "rb") as fin, zipfile.ZipFile(fin) as zip_file:
+            for filename in zip_file.namelist():
+                metadata_json = f"{item_identifier}/data.json"
+                if filename == metadata_json:
+                    return self._read_metadata_json_file(zip_file, metadata_json)
+            raise FileNotFoundError(
+                "The required file 'data.json' file was not found in the zip file: "
+                f"{file}"
+            )
+
+    def _read_metadata_json_file(
+        self, zip_file: zipfile.ZipFile, metadata_json: str
+    ) -> dict[str, str]:
+        """Read source metadata JSON file."""
+        with zip_file.open(metadata_json) as file:
+            source_metadata = json.load(file)
+            source_metadata["instructors"] = self._get_instructors_delimited_string(
+                source_metadata["instructors"]
+            )
+            return source_metadata
+
+    def _get_instructors_delimited_string(self, instructors: list[dict[str, str]]) -> str:
+        """Get delimited string of 'instructors' from source metadata JSON file.
+
+        Source metadata JSON files stored in OCW zip files contain an 'instructors'
+        property, which contains an array of objects representing an instructor's
+        credentials:
+
+            [
+                {
+                    "first_name": "Kerry",
+                    "last_name": "Oki",
+                    "middle_initial": "",
+                    "salutation": "Prof.",
+                    "title": "Prof. Kerry Oki"
+                },
+                {
+                    "first_name": "Earl",
+                    "last_name": "Bird",
+                    "middle_initial": "E.",
+                    "salutation": "Prof.",
+                    "title": "Prof. Earl E. Bird"
+                }
+            ]
+
+        Given these credentials, this method will construct a pipe-delimited ("|")
+        string with the following format: "<last_name>, <first_name> <middle_initial>".
+
+            Example output:
+                "Oki, Kerry|Bird, Earl E."
+
+        """
+        return "|".join(
+            [
+                instructor_name
+                for instructor in instructors
+                if (instructor_name := self._construct_instructor_name(instructor))
+            ]
+        ).strip()
+
+    @staticmethod
+    def _construct_instructor_name(instructor: dict[str, str]) -> str:
+        """Given a dictionary of name fields, derive instructor name."""
+        if not (last_name := instructor.get("last_name")) or not (
+            first_name := instructor.get("first_name")
+        ):
+            return ""
+        return f"{last_name}, {first_name} {instructor.get("middle_initial", "")}".strip()
```

Comment:
While it is plausible that all the metadata in `data.json` will always be formatted as needed (i.e., all `instructor` name fields provided), it would be a good idea to check in with stakeholders on the "minimum required `instructor` name fields` to construct an instructor name.

In this sample mapping file we received, [ocw_json_to_dspace_mapping.xlsx](https://github.com/user-attachments/files/18525047/ocw_json_to_dspace_mapping.xlsx), it indicates the `instructor` names must be formatted as: 

```text
<last_name>, <first_name> <middle_initial>
```

The code above will return an empty string if either the `last_name` or `first_name` is missing; it allows for missing `middle_initial` values.